functions/xbackstage/main.k

```diff
@@ -0,0 +1,145 @@
+import xeks
+import utils
+import ctpSecretSync
+
+oxr = sav1alpha1.XBackstage {**option("params").oxr}
+# observed composed resources
+_ocds = option("params").ocds
+# desired composite resource
+_dxr = option("params").dxr
+# desired composed resources
+dcds = option("params").dcds
+# extra resources
+_extra = option("params")?.extraResources
+
+_items = []
+
+if oxr.spec.parameters.infrastructure.eks.enabled:
+    _items += xeks.xeks(xeks.XEKS {
+        # shared
+        id: oxr.spec.parameters.id
+        providerConfigName: oxr.spec.parameters.providerConfigName
+        region: oxr.spec.parameters.region
+        deletionPolicy: oxr.spec.parameters.deletionPolicy
+        # xeks
+        accessConfig: oxr.spec.parameters.infrastructure.eks.accessConfig
+        iam: oxr.spec.parameters.infrastructure.eks.iam
+        nodes: oxr.spec.parameters.infrastructure.eks.nodes
+        version: oxr.spec.parameters.infrastructure.eks.version
+        uid: oxr.metadata.uid
+        # xnetwork
+        subnets: oxr.spec.parameters.infrastructure.network?.subnets
+        vpcCidrBlock: oxr.spec.parameters.infrastructure.network?.vpcCidrBlock
+    })
+
+_items += [
+    {
+        apiVersion: "meta.krm.kcl.dev/v1alpha1"
+        kind: "ExtraResources"
+        requirements = {
+            eks = {
+                apiVersion: "eks.aws.upbound.io/v1beta2"
+                kind: "Cluster"
+                matchLabels: {
+                    "crossplane.io/composite": oxr.metadata.name
+                }
+            },
+        }
+    }
+]
+
+if oxr.spec.parameters.infrastructure.eks.clusterName != "":
+    _eksClusterName = oxr.spec.parameters.infrastructure.eks.clusterName
+
+if _extra?.eks:
+    _eksClusterName = _extra?.eks[0]?.Resource?.metadata?.name or ""
+
+vpcId = _ocds["xnetwork"]?.Resource?.status?.vpcId
+
+if oxr.spec.parameters.infrastructure.eks.enabled or (oxr.spec.parameters.infrastructure.eks.enabled == False and _eksClusterName != ""):
+
+    if vpcId and _eksClusterName != Undefined:
+        _items += lbcontroller.lbcontroller(lbcontroller.LBController {
+            clusterName: _eksClusterName
+            name: oxr.metadata.name
+            vpcId: vpcId
+            id: oxr.spec.parameters.id
+            providerConfigName: oxr.spec.parameters.providerConfigName
+            region: oxr.spec.parameters.region
+            deletionPolicy: oxr.spec.parameters.deletionPolicy
+            ocds: _ocds
+            dependsOn: "xeks"
+        })
+
+    if oxr.spec.parameters.infrastructure.certManager.enabled:
+        _items += certmanager.certManager(certmanager.CertManager {
+            name: oxr.metadata.name
+            id: oxr.spec.parameters.id
+            deletionPolicy: oxr.spec.parameters.deletionPolicy
+            ocds: _ocds
+            dependsOn: "{}-lb-controller".format(oxr.metadata.name)
+        })
+
+
+    if oxr.spec.parameters.infrastructure.ingressNginx.enabled and _eksClusterName != Undefined:
```

Comment:
As a nit we could wrap all the ones which require the cluster name together, but also ok like it is atm